### PR TITLE
core: make sure ProtectHostname= is handled gracefully in containers …

### PR DESCRIPTION
**Pull request by poettering**
…lacking seccomp

Fixes: #15408

### CODE REVIEW

File: src/core/execute.c [hunk 1]
```diff
@@ -1577,7 +1577,40 @@ static int apply_protect_kernel_logs(const Unit *u, const ExecContext *c) {
...
+                        }
+
+                        log_unit_warning(u, "ProtectHostname=yes is configured, but UTS namespace setup is prohibited (container manager?), ignoring namespace setup.");
+                }
+        } else
+                log_unit_warning(u, "ProtectHostname=yes is configured, but the kernel does not support UTS namespaces, ignoring namespace setup.");
+
+        if (skip_seccomp_unavailable(u, "ProtectHostname="))
+                return 0;
+
```
Comment (topimiettinen) [inline src/core/execute.c]: Should keep `#if HAVE_SECCOMP`.

File: src/core/execute.c [hunk 2]
```diff
@@ -1577,7 +1577,40 @@ static int apply_protect_kernel_logs(const Unit *u, const ExecContext *c) {
...
+        r = seccomp_protect_hostname();
+        if (r < 0) {
+                *ret_exit_status = EXIT_SECCOMP;
+                return log_unit_error_errno(u, r, "Failed to apply hostname restrictions: %m");
+        }
+
```
Comment (topimiettinen) [inline src/core/execute.c]: ``#endif``

Comment (topimiettinen) [reply src/core/execute.c]: LGTM otherwise.

File: src/core/execute.c (hunk 1 above)
Comment (poettering) [reply src/core/execute.c]: indeed. will fix.

**Comment by topimiettinen** [comment]
LGTM

**Pull request merged**